### PR TITLE
[framework] eased adding JS to admin

### PR DESCRIPTION
**Pull request by simara-svatopluk**
| Q             | A
| ------------- | ---
|Description, reason for the PR| it used to be difficult to add JS to admin instead of rewriting all admin JS
|New feature| Yes
|BC breaks| No
|Fixes issues| ...
|Standards and tests pass| Yes
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

**Comment by simara-svatopluk** [comment]
This is unfortunately a hotfix solution, because `JavascriptCompilerService` has horrible way of processing javascripts

File: docs/introduction/framework-extensibility.md [hunk 1]
```diff
@@ -22,6 +22,8 @@ as well as a list of customizations that are not (and will not be) possible at a
 * open-box modifications in `project-base`
     * e.g. adding new entities, changing the FE design, customization of FE javascripts, adding new FE pages (routes and controllers), ...
 * [Hiding the existing features and functionality](https://github.com/shopsys/demoshop/pull/13)
+* add javascript into admin
```
Comment (Miroslav-Stopka) [inline docs/introduction/framework-extensibility.md]: "add javascript into admin" -> "adding a new javascript into admin" (see previous entries with adding)

File: docs/introduction/framework-extensibility.md [hunk 2]
```diff
@@ -22,6 +22,8 @@ as well as a list of customizations that are not (and will not be) possible at a
...
+    * add it into directory `src/Shopsys/ShopBundle/Resources/scripts/custom_admin` and it will be loaded automatically
```
Comment (Miroslav-Stopka) [inline docs/introduction/framework-extensibility.md]: add the new javascript files into the directory `src/Shopsys/ShopBundle/Resources/scripts/custom_admin` and they will be

File: packages/framework/src/Twig/Javascript/JavascriptCompilerService.php [hunk 3]
```diff
@@ -148,7 +148,7 @@ private function getPathWithTimestamp($relativePath, $timestamp)
     private function getRelativeTargetPath($javascript)
     {
         $relativeTargetPath = null;
-        if (strpos($javascript, 'admin/') === 0 || strpos($javascript, 'frontend/') === 0 || strpos($javascript, 'common/') === 0) {
+        if (strpos($javascript, 'admin/') === 0 || strpos($javascript, 'frontend/') === 0 || strpos($javascript, 'common/') === 0 || strpos($javascript, 'custom_') === 0) {
```
Comment (Miroslav-Stopka) [inline packages/framework/src/Twig/Javascript/JavascriptCompilerService.php]: You are using directory "custom_admin" everywhere but here there is checking for "custom_". Why?

**Comment by Miroslav-Stopka** [comment]
Information for tester: 

1. scenario - new folder in ShopBundle/Resources/scripts/ with the name admin -> ShopBundle/Resources/scripts/admin/ . Is administration javascripts still functional?

2. scenario - new js file in ShopBundle/Resources/scripts/custom_admin/, is this js file automatically loaded in administration? And only in administration?

3. scenario - new directory with js file in ShopBundle/Resources/scripts/custom_admin/ -> ShopBundle/Resources/scripts/custom_admin/something/123.js , is this js file automatically loaded in administration? And only in administration?

Please test these scenarios.

File: packages/framework/src/Twig/Javascript/JavascriptCompilerService.php (hunk 3 above)
Comment (simara-svatopluk) [reply packages/framework/src/Twig/Javascript/JavascriptCompilerService.php]: When I was discovering how it works, I added a line `custom_admin/*.js` into twig, and nothing happend. So I was debugging, and found this condition. I have no idea why the condition is here and why is so strict, it just restricted me from adding the javascript.

So, this is not as strict as previous ones and can be used for example for "custom_frontend"

Comment (Miroslav-Stopka) [reply packages/framework/src/Twig/Javascript/JavascriptCompilerService.php]: Still not answered, why there is checking for "custom_" and not for "custom_admin"?

**Comment by Miroslav-Stopka** [comment]
Hi @simara-svatopluk,
CR done, sending to TEST. Nice done :)

**Comment by MattCzerner** [comment]
Hey, i tested it out and it all works fine. I wouldn't name the folder admin_custom. In project-base we got Controllers for admin and we dont name it AdminCustom... But actually thats your choice i wouldnt mind have it named like it is now.

**Comment by PetrHeinz** [comment]
@MattCzerner: How would you name it? I can think of `autoload_admin` / `admin_autoload`, but it's not much better...

**Comment by simara-svatopluk** [comment]
In the end I find name `admin_custom` suitable

**Pull request merged**